assets/src/js/admin/reports/components/mini-period-selector/index.js

```diff
@@ -0,0 +1,32 @@
+// Store-related dependencies
+import { useStoreValue } from '../../store';
+
+import './style.scss';
+const { __ } = wp.i18n;
+
+const MiniPeriodSelector = () => {
+	// Get 'period' object from the store
+	const [ { period }, dispatch ] = useStoreValue();
+
+	// Dispatch SET_RANGE action
+	const setRange = ( range ) => {
+		dispatch( {
+			type: 'SET_RANGE',
+			payload: {
+				range,
+			},
+		} );
+	};
```

Comment:
Reducer action functions should be stored with or near the context. If the reducer is simple enough I'll often just store them in the same file. Then you don't even have to import the dispatch function.

Reply:
Done, see ref: https://github.com/impress-org/givewp/commit/87948ba4b95241d7579efcd3cbed33b53fe4c119